docs/src/explanations/dpi.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: docs
+title:  "Calling Native Functions from Chisel (DPI)"
+section: "chisel3"
+---
+
+# Calling Native Functions from Chisel (DPI)
+
+## DPI Basics
+
+DPI API allows you to integrate native code into your Chisel hardware designs. This enables you to leverage existing libraries or implement functionality that is difficult to express directly in Chisel.
+
+Here's a simple example that demonstrates printing a message from a C++ function:
+```c++
+extern "C" void hello()
+{
+    std::cout << "hello from c++\\n";
+}
+```
+
+To call this function from Chisel, we need to define a corresponding DPI object.
+
+```scala
+object Hello extends DPIVoidFunctionImport {
+  override val functionName = "hello"
+  override val clocked = true
+  final def apply() = super.call()
+}
+
+Hello() // Print
```

Comment:
should this `Hello()` call be inside a Module?

Reply:
Yes. The snippet was meant to be pseudo code so I'll update with mdoc.